x-pack/solutions/observability/plugins/slo/server/services/get_slo_stats_overview.ts

```diff
@@ -31,10 +46,143 @@ export class GetSLOStatsOverview {
+      });
+    }
+
+    const instanceIdIncluded = Object.values(params).find(
+      (value) => typeof value === 'string' && value.includes('slo.instanceId')
+    );
+
+    try {
+      if (querySLOsForIds) {
+        do {
+          const sloIdCompositeQueryResponse = await this.scopedClusterClient.asCurrentUser.search({
+            size: 0,
+            aggs: {
+              sloIds: {
+                composite: {
+                  after: afterKey,
+                  size: ES_PAGESIZE_LIMIT,
+                  sources: [
+                    {
+                      sloId: { terms: { field: 'slo.id' } },
+                    },
+                    ...(instanceIdIncluded
+                      ? [
+                          {
+                            sloInstanceId: { terms: { field: 'slo.instanceId' } },
+                          },
+                        ]
+                      : []),
+                  ],
+                },
+              },
+            },
+            index: '.slo-observability.summary-*',
+            _source: ['slo.id', 'slo.instanceId'],
+            ...(Object.values(boolFilters).some((value) => Array.isArray(value) && value.length > 0)
+              ? {
+                  query: {
+                    bool: boolFilters,
+                  },
+                }
+              : {}),
+          });
+
+          totalHits = (sloIdCompositeQueryResponse.hits?.total as SearchTotalHits).value || 0;
+          afterKey = getAfterKey(sloIdCompositeQueryResponse.aggregations?.sloIds);
+
+          const buckets = (
+            sloIdCompositeQueryResponse.aggregations?.sloIds as {
+              buckets?: Array<{ key: { sloId: string; sloInstanceId: string } }>;
+            }
+          )?.buckets;
+          if (buckets && buckets.length > 0) {
+            sloKeysFromES = sloKeysFromES.concat(
+              ...buckets.map((bucket) => {
+                sloRuleKeysFromES.push(bucket.key.sloId);
+                return {
+                  bool: {
+                    must: [
+                      { term: { 'kibana.alert.rule.parameters.sloId': bucket.key.sloId } },
+                      ...(instanceIdIncluded
+                        ? [
+                            {
+                              term: {
+                                'kibana.alert.instance.id': bucket.key.sloInstanceId,
+                              },
+                            },
+                          ]
+                        : []),
+                    ],
+                  },
+                };
+              })
+            );
+          }
+        } while (afterKey);
+
+        const sloIdsArray = Array.from(sloRuleKeysFromES);
+
+        const resultNodes =
+          sloIdsArray.length > 0
+            ? nodeBuilder.or(
+                sloIdsArray.map((sloId) => nodeBuilder.is(`alert.attributes.params.sloId`, sloId))
+              )
+            : nodeBuilder.is(`alert.attributes.params.sloId`, '%NO%MATCHES%');
```

Comment:
```suggestion
            : nodeBuilder.is(`alert.attributes.params.sloId`, '%NO%MATCHES%');
```

~Showing my lack of SLO knowledge here, is `%NO%MATCHES%` some magic identifier?~

After talking offline I understand now why this is here. I'll try to work with you to work around needing to do this as I suspect we can simplify here.